### DISCUSSION TITLE
Create an app programmatically

### DISCUSSION THREAD

**Discussion by kerk1v** [caprover/caprover#1846] [Q&A]
Hello, 

To improve my workflow, I would like to create or update apps (including all parameters, Env Variables, HTTP(S) config, etc... programatically, via an API call or something similar. I would / could also use the CLI if only the API function was a bit better documented. 

I looked at https://pypi.org/project/caprover-api/ but it seems to be outdated and unmaintained and throws errors. 

Can anyone point me in the right direction? 

Thanks!

**Comment by lexoyo** [comment]
Hello
I do this with the cli, it works pretty well
It comes down to `node_modules/.bin/caprover deploy -b $CAPROVER_BRANCH --caproverName $CAPROVER_NAME -a $APP_NAME -p $CAPROVER_PASSWORD`

**Reply by lexoyo** [reply]
also i update env vars and settings with calls like this: `node_modules/.bin/caprover api --caproverName $CAPROVER_NAME -p $CAPROVER_PASSWORD -m POST -t /user/apps/appDefinitions/enablebasedomainssl -d "{\"appName\":\"$APP_NAME\"}"`
